sqlair.go

```diff
@@ -51,17 +51,19 @@ func NewDB(db *sql.DB) *DB {
 	return &DB{db: db}
 }
 
-// Unwrap returns the underlying database object.
-func (db *DB) Unwrap() *sql.DB {
+// PlainDB returns the underlying database object.
+func (db *DB) PlainDB() *sql.DB {
 	return db.db
 }
 
 // Query holds the results of a database query.
 type Query struct {
-	qe   *expr.QueryExpr
-	q    func() (*sql.Rows, error)
-	rows *sql.Rows
-	err  error
+	qe      *expr.QueryExpr
+	outcome *Outcome
+	closed  bool
+
+	q func() (*sql.Rows, error)
+	e func() (sql.Result, error)
```

Comment:
That's the opposite of what we discussed recently. We should remove the closure altogether, rather than adding one more.

Also, how can a query be closed when it was never executed?  The Close method only exists on the Iterator.

How can it have an outcome?  Similar rationale..

Something seems quite off here, or I misunderstand is the idea.

Reply:
After trying out the alternatives I stuck with the closures because it seemed like a really nice way to make the Query object agnostic to whether it was running on a transaction or straight on the database. I didn't find any other options that felt as good but if there are any better ways that would be good.

The close `bool` is to stop people accidentally running the query twice. Once the underlying `Rows` are closed then there is no way to open then again and if we were to allow them to run `One` again this would have to run the query again, which might not be expected. Once a method has been run on the query it gets closed and nothing else can be run on it.

It seems I may have misunderstood exactly what you were hoping for with the `Outcome`.